tools/bumper/bumper.go

```diff
@@ -0,0 +1,194 @@
+		_, err = getRepositoryOnBranch(repoDir, component.Url, component.Branch)
+		if err != nil {
+			exitWithError(errors.Wrapf(err, "Failed to clone %s", componentName))
+		}
+
+		currentReleaseTag, err := gitMgr.getCurrentReleaseTag(component, repoDir)
+		if err != nil {
+			exitWithError(errors.Wrapf(err, "Failed to get current release version tag from %s", componentName))
+		}
+
+		updatedReleaseTag, updatedReleaseCommit, err := gitMgr.getUpdatedReleaseInfo(component, repoDir)
+		if err != nil {
+			exitWithError(errors.Wrapf(err, "Failed to get latest release version tag from %s", componentName))
+		}
+
+		bumpNeeded, err := isBumpNeeded(currentReleaseTag, updatedReleaseTag, component.Updatepolicy)
+		if err != nil {
+			exitWithError(errors.Wrapf(err, "Failed to discover if Bump need for %s", componentName))
+		}
+
+		if bumpNeeded {
+			//reset --hard git repo
+			exitWithError(fmt.Errorf("reset --hader repo not implemented yet"))
+
+			//PR name
+			PRTitle := fmt.Sprintf("Bump %s to %s", componentName, updatedReleaseTag)
+			logger.Printf("PR title: %s", PRTitle)
+			//Create PR
+			exitWithError(fmt.Errorf("create PR not implemented yet"))
+
+			//update component's entry in config yaml
+			component.Commit = updatedReleaseCommit
+			component.Metadata = updatedReleaseTag
+			err = updateComponentsYaml(inputArgs.componentsConfigPath, componentsConfig)
+			if err != nil {
+				exitWithError(errors.Wrap(err, "Failed to update components yaml"))
+			}
+
+			// run bump script
+			cmd := exec.Command("make", fmt.Sprintf("bump-%s", componentName))
+			if out, err := cmd.CombinedOutput(); err != nil {
+				exitWithError(errors.Wrapf(err, "Failed to run bump script. StdOut = %s", string(out)))
+			}
+
+			//create a new branch name
+			BranchName := strings.Replace(strings.ToLower(PRTitle), " ", "_", -1)
+			logger.Printf("Opening new Branch %s", BranchName)
+
+			//push branch to PR
+			exitWithError(fmt.Errorf("push branch to PR not implemented yet"))
+		}
+	}
+}
+
+func getRepositoryOnBranch(repoDir, repositoryURL string, branchName string) (*git.Repository, error) {
+	logger.Printf("Cloning to temp directory: %s", repoDir)
+
+	return git.PlainClone(repoDir, false, &git.CloneOptions{
+		URL:           repositoryURL,
+		ReferenceName: plumbing.NewBranchReferenceName(branchName),
+	})
+}
+
+func isBumpNeeded(currentReleaseVersion, latestReleaseVersion, updatePolicy string) (bool, error) {
+	logger.Printf("currentReleaseVersion: %s, latestReleaseVersion: %s, updatePolicy: %s\n", currentReleaseVersion, latestReleaseVersion, updatePolicy)
+
+	if updatePolicy == updatePolicyStatic {
+		logger.Printf("updatePolicy is static. avoiding auto bump")
+		return false, nil
+	}
+
+	//if one of the tags is in vtag format (e.g 0.39.0-32-g1fcbe815), and not equal, then always bump
+	if isVtagFormat(currentReleaseVersion) || isVtagFormat(latestReleaseVersion) {
+		return currentReleaseVersion == latestReleaseVersion, nil
+	} else {
```

Comment:
No need for else branch. There is an early return handled by the if. You can continue with the rest of the code on the root of the function.

The idiomatic way here is that the default flow should happen with as little indentation as possible.

Reply:
Done